src/Widgets/ScreenBrightness.vala

```diff
@@ -54,9 +54,15 @@ public class Power.Widgets.ScreenBrightness : Gtk.EventBox {
         grid.add (image);
         grid.add (brightness_slider);
 
-        add (grid);
+        var show_brightness_slider = new Gtk.Revealer ();
+        show_brightness_slider.add (grid);
 
-        brightness_slider.set_value (dm.brightness);
+        add (show_brightness_slider);
+
+        if (dm.brightness != -1) {
```

Comment:
isn't there an other way to verify that the device doesn't have brightness control?  this seems a little hacky IMO

Reply:
I didnt found other way. Maybe creating a property ?

Reply:
That would be nice to see. Dm.brigthness.control_present (boolean) instead.

Reply:
Unless that property is going to be used in several places its probably not worth creating a new entity - I would just add an explanatory comment.